gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+        return ev
+
+    def _cancelled(self, eh, ev):
+        LOG.info(_(
+            "Event %s cancelled -"
+            "invoking %s handler's poll_event_cancel method"
+            % (ev.identify(), identify(eh))))
+        try:
+            self._sc.poll_event_done(ev)
+            eh.poll_event_cancel(ev)
+        except AttributeError:
+            LOG.info(_(
+                "Handler %s does not implement"
+                "poll_event_cancel method" % (identify(eh))))
+
+    def _sched(self, eh, ev):
+        LOG.debug(_(
+            "Event %s to be scheduled to handler %s"
+            % (ev.identify(), identify(eh))))
+        if isinstance(eh, core_periodic_task.PeriodicTasks):
+            peh = eh.get_periodic_event_handler(ev)
+            if peh:
+                t = self._tpool.dispatch(peh, eh, ev)
+                LOG.info(_(
+                    "Dispatched method %s of handler %s"
+                    "for event %s to thread %s"
+                    % (identify(peh), identify(eh),
+                        ev.identify(), t.identify())))
+
+            else:
+                t = self._tpool.dispatch(eh.handle_poll_event, ev)
+                LOG.info(_(
+                    "Dispatched handle_poll_event() of handler %s"
+                    "for event %s to thread %s"
+                    % (identify(eh),
+                        ev.identify(), t.identify())))
+        else:
+            self._tpool.dispatch(eh.handle_poll_event, ev)
+            LOG.info(_(
+                "Dispatched handle_poll_event() of handler %s"
+                "for event %s to thread %s"
+                % (identify(eh),
+                    ev.identify(), t.identify())))
+
+    def run(self, qu):
+        LOG.info(_("Started worker process - %s" % (process)))
+        while True:
+            ev = self._get()
+            if ev:
+                LOG.debug(_("Got event %s" % (ev.identify())))
+                eh = self._eh.get(ev)
+                if not ev.poll_event:
+                    t = self._tpool.dispatch(eh.handle_event, ev)
+                    LOG.debug(_(
+                        "Event %s is not poll event - "
+                        "disptaching handle_event() of handler %s"
+                        "to thread %s"
+                        % (ev.identify(), identify(eh), t.identify())))
+                else:
+                    if ev.poll_event == 'POLL_EVENT_CANCELLED':
+                        LOG.debug(
+                            _("Got cancelled event %s" % (ev.identify())))
+                        self._cancelled(eh, ev)
+                    else:
+                        LOG.info(
+                            _("Got POLL Event %s scheduling"
+                                % (ev.identify())))
+                        self._sched(eh, ev)
+            time.sleep(0)  # Yield the CPU
+
+
+class EventHandlers(object):
+
+    def __init__(self):
+        self._ehs = {}
+
+    def register(self, ev):
```

Comment:
Rename 'ev' to something like event_desc or event_info, to differentiate from an event in the queue.